django/db/models/sql/compiler.py

```diff
@@ -657,7 +674,18 @@ def get_qualify_sql(self):
                 ")",
                 self.connection.ops.quote_name("qualify_mask"),
             ]
-        return result, list(inner_params) + qualify_params
+        params = list(inner_params) + qualify_params
+        # As the SQL spec is unclear on whether or not derived tables
+        # ordering must propagate it has to be explicitly repeated on the
+        # outer-most query to ensure it's preserved.
+        if order_by:
+            ordering_sqls = []
+            for ordering in order_by:
+                ordering_sql, ordering_params = self.compile(ordering)
+                ordering_sqls.append(ordering_sql)
+                ordering_params.extend(ordering_params)
```

Comment:
`ordering_params` are overwritten and are not used anywhere :thinking: As far as I'm aware, we should add them to `params`
```suggestion
                ordering_sql, ordering_params = self.compile(ordering)
                ordering_sqls.append(ordering_sql)
                params.extend(ordering_params)
```

Reply:
You are completely right, I guess we don't have any test covering ordering by a function hence why tests were not failing here.